OSDOCS-10363 RN for AUTH-509 (#77075)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

* `Standard`
* `Slower`

[discrete]
[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]

> === Unauthenticated users no longer have access to cluster role bindings

Reviewer: ### Role Bindings vs Roles

The `system:unauthenticated` was removed from cluster role bindings, which leads to the loss of access that was given by the cluster roles. So something like this would be more correct:

```
Unauthenticated users were removed from some access to cluster roles.
```

OR

```
Unauthenticated users were removed from some cluster role bindings.
```

### Unauthenticated still have access to certain roles

`system:unauthenticated` wasn't removed from access to all roles. We kept unauthenticated in two "public" roles. They contain public information that are also being exposed by default in kubernetes upstream.

So we could emphasize that we removed or narrowed down unnecessary access for unauthenticated groups or that we removed general access, which is only being used in certain use cases?

GitHub is a use-case, where you need to give up the k8s / ocp authentication as they don't support it. So users who would not want GitHub to trigger builds, had unnecessarily access for unauthenticated.

This isn't a vulnerability, as there is a custom solution by GitHub, that ensures authenticity. But the custom solution needs to by-pass k8s / ocp auth solutions. 
The solution from GitHub relies on a shared secret, that is being used to create a hash, that needs to be validated.

Reviewer: #### Reasons

##### For removal

- Defense in Depth
- Isn't necessary for a default installation

##### Why it was there

- Lack of K8s support for Feature-Providers like GitHub.

> In {product-title} {product-version} unauthenticated users will no longer have default access to cluster role bindings. This change is for new clusters, previous clusters will continue to allow unauthenticated access to cluster role bindings.

Reviewer: #### Affects only a 4 ClusterRoleBindings

 would narrow it down:

"to specific cluster role bindings that are necessary for certain feature sets".

We should avoid to give the impression, that unauthenticated have access to a broad range of cluster roles.

#### Role vs Role Binding

"Access to a role" or "Being part of a role binding".

> In particular, this change impacts `BuildConfigs` processes. The `system:webhook` role allows users to trigger builds from external systems such as GitHub, GitLab, and Bitbucket. Unauthenticated users can still do this if granted access to the `system:webhook` cluster role binding by a cluster administrator.

Reviewer: The access would be to the cluster role `system:webhooks`. This can be done by a (cluster) role binding. As mentioned in the Jira ticket, it would be preferred, if the administrator would use role bindings, instead of cluster role bindings. It would narrow down the access to certain namespace, which is a great improvement. But I would like to test if this really works 😄

* `Standard`
* `Slower`

[discrete]
[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]
=== Unauthenticated users were removed from some cluster role bindings.

> In {product-title} {product-version} unauthenticated users will no longer have access to most cluster role bindings. Changing this access for unathenticated users adds an additional layer of security and will not impact most users. This change is for new clusters, previous clusters will not be effected.

Reviewer 2: ```suggestion
With this release, unauthenticated users no longer have access to most cluster role bindings. Changing this access for unauthenticated users adds an additional layer of security and does not impact most users. This change is for new clusters; previous clusters are not affected.
```

> There are use cases where allowing unauthenticated users specific cluster role bindings is desired. For example, the `system:webhook` cluster role allows users to trigger builds from external systems such as GitHub, GitLab, and Bitbucket. Unauthenticated users can still do this if granted access to the `system:webhook` cluster role binding by a cluster administrator.

Reviewer 2: Maybe reorder the last sentence to be more active voice - start with saying that cluster admins can grant access...

> To grant unauthenticated users access to specific cluster role bindings that are necessary for certain feature sets, see xref:../modules/unauthenticated-users-cluster-role-binding.adoc#modules-unauth-cluster-role-bindings[Adding unauthenticated groups to cluster role bindings].

Reviewer 2: I'd be careful of using "feature sets" unless we specifically mean the feature set / feature gate things in OCP. Which I don't _believe_ you mean here. If you don't mean those, then I'd just go with "features".

* `Standard`
* `Slower`

[discrete]
[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]
=== Unauthenticated users were removed from some cluster role bindings

> With this release, unauthenticated users no longer have access to most cluster role bindings. Before {product-title} {product-version} unauthenticated users could access cluster role bindings. Changing this access for unauthenticated users adds an additional layer of security and does not impact most users. This change is for new clusters; previous clusters are not affected.

Reviewer: ```suggestion
With this release, unauthenticated users no longer have access to most cluster role bindings. Before {product-title} {product-version} unauthenticated users could access certain cluster role bindings. Changing this access for unauthenticated users adds an additional layer of security and does not impact most users. This change is for new clusters; previous clusters are not affected.
```

There are use cases where allowing unauthenticated users specific cluster role bindings is desired. To grant unauthenticated users access to specific cluster role bindings that are necessary for certain features, see xref:../modules/unauthenticated-users-cluster-role-binding.adoc#modules-unauth-cluster-role-bindings[Adding unauthenticated groups to cluster role bindings].

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

[discrete]
[id="ocp-4-16-unauthenticated-users-system-webhook-role-binding_{context}"]
=== Unauthenticated users were removed from the `system:webhook` role binding

> With this release, unauthenticated users no longer have access to the `system:webhook` role binding. Before {product-title} {product-version} unauthenticated users could access the `system:webhook` role binding. Changing this access for unauthenticated users adds an additional layer of security and does not impact most users. This change is for new clusters; previous clusters are not affected.

Reviewer: ```suggestion
With this release, unauthenticated users no longer have access to the `system:webhook` role binding. Before {product-title} {product-version} unauthenticated users could access the `system:webhook` role binding. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled for users in need. This change is for new clusters; previous clusters are not affected.
```

I don't know how many do use that feature, but as long as it doesn't apply nearly everyone, it makes sense to make it opt-in.

* `Standard`
* `Slower`

[discrete]
[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]
=== Unauthenticated users were removed from some cluster role bindings

> With this release, unauthenticated users no longer have access to specific cluster role bindings that are necessary for certain feature sets. Before {product-title} {product-version} unauthenticated users could access certain cluster role bindings. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by users when necessary. This change is for new clusters; previous clusters are not affected.

Reviewer: ```suggestion
With this release, unauthenticated users no longer have access to specific cluster roles that are necessary for certain feature sets. Before {product-title} {product-version} unauthenticated users could access certain cluster roles. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by users when necessary. This change is for new clusters; previous clusters are not affected.
```

> There are use cases where allowing unauthenticated users specific cluster role bindings is desired. To grant unauthenticated users access to specific cluster role bindings that are necessary for certain features, see xref:../modules/unauthenticated-users-cluster-role-binding.adoc#modules-unauth-cluster-role-bindings[Adding unauthenticated groups to cluster role bindings].

Reviewer: ```suggestion
There are use cases where giving access to unauthenticated users for specific cluster roles is desired. To grant unauthenticated users access to specific cluster roles that are necessary for certain features, see xref:../modules/unauthenticated-users-cluster-role-binding.adoc#modules-unauth-cluster-role-bindings[Adding unauthenticated groups to cluster role bindings].
```

* `Standard`
* `Slower`

[discrete]
[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]
=== Unauthenticated users were removed from some cluster roles

> With this release, unauthenticated users no longer have access to specific cluster roles that are necessary for certain feature sets. Before {product-title} {product-version} unauthenticated users could access certain cluster roles. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by users when necessary. This change is for new clusters; previous clusters are not affected.

Reviewer 3: I tend to be comma-happy. Do we need one here?
```suggestion
With this release, unauthenticated users no longer have access to specific cluster roles that are necessary for certain feature sets. Before {product-title} {product-version}, unauthenticated users could access certain cluster roles. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by admin (??) when necessary. This change is for new clusters; previous clusters are not affected.
```

> There are use cases where giving access to unauthenticated users for specific cluster roles is desired. To grant unauthenticated users access to specific cluster roles that are necessary for certain features, see xref:../modules/unauthenticated-users-cluster-role-binding.adoc#modules-unauth-cluster-role-bindings[Adding unauthenticated groups to cluster role bindings].

Reviewer 3: ISG says not to use "desire," but use "want." 
```suggestion
There are use cases where you might want to give access to unauthenticated users for specific cluster roles. To grant unauthenticated users access to specific cluster roles that are necessary for certain features, see xref:../modules/unauthenticated-users-cluster-role-binding.adoc#modules-unauth-cluster-role-bindings[Adding unauthenticated groups to cluster role bindings].
```

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

[discrete]
[id="ocp-4-16-unauthenticated-users-system-webhook-role-binding_{context}"]
=== Unauthenticated users were removed from the `system:webhook` role binding

> With this release, unauthenticated users no longer have access to the `system:webhook` role binding. Before {product-title} {product-version} unauthenticated users could access the `system:webhook` role binding. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by users when necessary. This change is for new clusters; previous clusters are not affected.

Reviewer 3: ```suggestion
With this release, unauthenticated users no longer have access to the `system:webhook` role binding. Before {product-title} {product-version}, unauthenticated users could access the `system:webhook` role binding. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by users when necessary. This change is for new clusters; previous clusters are not affected.
```

> There are use cases where allowing unauthenticated users the `system:webhook` role binding for specific namespaces is desired. The `system:webhook` cluster role allows users to trigger builds from external systems that do not use {product-title} authentication  mechanisms, such as GitHub, GitLab, and Bitbucket. Cluster admins can grant unauthenticated users access to the `system:webhook` role binding to facilitate this use case.

Reviewer 3: ```suggestion
There are use cases where you might want to allow unauthenticated users the `system:webhook` role binding for specific namespaces. The `system:webhook` cluster role allows users to trigger builds from external systems that do not use {product-title} authentication  mechanisms, such as GitHub, GitLab, and Bitbucket. Cluster admins can grant unauthenticated users access to the `system:webhook` role binding to facilitate this use case.
```

There are use cases where you might want to allow unauthenticated users the `system:webhook` role binding for specific namespaces. The `system:webhook` cluster role allows users to trigger builds from external systems that do not use {product-title} authentication  mechanisms, such as GitHub, GitLab, and Bitbucket. Cluster admins can grant unauthenticated users access to the `system:webhook` role binding to facilitate this use case.

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

To grant unauthenticated users access to the `system:webhook` role binding in specific namespaces, see xref:../../triggering-builds-build-hooks.adoc#unauthenticated-users-system-webhook_triggering-builds-build-hooks[Adding unauthenticated users to the system:webhook role binding].

> 

Reviewer 4: Should this be moved to the "Build" section of "New features and enhancements"? I'd call AUTH-509 a security enhancement, and moving it in the "Build" section places the impact front and center. I don't think any other component in OCP uses the "system:webhook" role directly.

Author: @adambkaplan I can move it there if you think that's a better place for customers to find it.

Reviewer 4: I think that will help customers discover the change more easily - thanks!

Author: @adambkaplan I've moved it. The preview takes a few minutes to update with the change if you want to take a look.

[id="ocp-4-16-builds_{context}"]
=== Builds

[discrete]
[id="ocp-4-16-unauthenticated-users-system-webhook-role-binding_{context}"]
=== Unauthenticated users were removed from the system:webhook role binding

With this release, unauthenticated users no longer have access to the `system:webhook` role binding. Before {product-title} {product-version}, unauthenticated users could access the `system:webhook` role binding. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled by users when necessary. This change is for new clusters; previous clusters are not affected.

There are use cases where you might want to allow unauthenticated users the `system:webhook` role binding for specific namespaces. The `system:webhook` cluster role allows users to trigger builds from external systems that do not use {product-title} authentication  mechanisms, such as GitHub, GitLab, and Bitbucket. Cluster admins can grant unauthenticated users access to the `system:webhook` role binding to facilitate this use case.

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

> To grant unauthenticated users access to the `system:webhook` role binding in specific namespaces, see xref:../../triggering-builds-build-hooks.adoc#unauthenticated-users-system-webhook_triggering-builds-build-hooks[Adding unauthenticated users to the system:webhook role binding].

Reviewer 5: ```suggestion
To grant unauthenticated users access to the `system:webhook` role binding in specific namespaces, see xref:../cicd/builds/triggering-builds-build-hooks.adoc#unauthenticated-users-system-webhook_triggering-builds-build-hooks[Adding unauthenticated users to the system:webhook role binding].
```

* `Standard`
* `Slower`

[discrete]
[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]
=== Unauthenticated users were removed from some cluster roles

With this release, unauthenticated users no longer have access to specific cluster roles that are necessary for certain feature sets. Before {product-title} {product-version} unauthenticated users could access certain cluster roles. Changing this access for unauthenticated users adds an additional layer of security and should only be enabled when necessary. This change is for new clusters; previous clusters are not affected.

> There are use cases where you might want to give access to unauthenticated users for specific cluster roles. To grant unauthenticated users access to specific cluster roles that are necessary for certain features, see xref:../../post_installation_configuration/preparing-for-users.adoc#unauthenticated-users-cluster-role-bindings_post-install-preparing-for-users[Adding unauthenticated groups to cluster roles]

Reviewer 5: ```suggestion
There are use cases where you might want to give access to unauthenticated users for specific cluster roles. To grant unauthenticated users access to specific cluster roles that are necessary for certain features, see xref:../post_installation_configuration/preparing-for-users.adoc#unauthenticated-users-cluster-role-bindings_post-install-preparing-for-users[Adding unauthenticated groups to cluster roles]
```
